packages/EasyApiToken/src/External/AwsCognitoJwtDriver.php

```diff
@@ -48,8 +56,16 @@ public function decode(string $token): object
             JWT::$leeway = $this->leeway;
         }
 
-        /** @var \stdClass $decodedToken */
-        $decodedToken = JWT::decode($token, $this->jwkFetcher->getJwks($this->userPoolConfig), $this->allowedAlgos);
+        $jwks = $this->jwkFetcher->getJwks($this->userPoolConfig);
+
+        if (self::isFirebaseJwtV5() === false) {
```

Comment:
This check looks like we are OK with versions <= 4 in this `if`. Let's either do `isFirebaseJwtV5` -> `getFirebaseJwtVersion` or `isFirebaseJwtV5` -> `isFirebaseJwtVersionGt5`. Or anything like that. But I don't like the 2nd option actually, the name looks strange to me.

Reply:
Refactored, please check.